src/renderer/main.tsx

```diff
@@ -61,9 +62,37 @@ export const Main = memo(() => {
         )
     );
 
+    const [status, setStatus] = useState(t('splash.loading', 'Loading...'));
+    const [statusProgress, setStatusProgress] = useState<null | number>(null);
+    const [overallProgress, setOverallProgress] = useState(0);
+
+    // Register event listeners
+    useEffect(() => {
+        ipcRenderer.on('setup-progress', (event, progress) => {
+            setOverallProgress(progress.totalProgress);
+            setStatusProgress(progress.statusProgress > 0 ? progress.statusProgress : null);
+
+            if (progress.status) {
+                setStatus(progress.status);
+            }
+        });
+    }, []);
+
+    useEffect(() => {
+        ipcRenderer.send(isBackendReady ? 'enable-menu' : 'disable-menu');
```

Comment:
I don't like the idea that the menu starts out enabled, then gets disabled shortly after, only to be re-enabled when the backend is ready.

Could we start with the menu disabled instead and skip the disabling from here?